sklearn/utils/validation.py

```diff
@@ -547,14 +555,44 @@ def check_array(array, accept_sparse=False, dtype="numeric", order=None,
         msg = ("Data with input dtype %s was converted to %s%s."
                % (dtype_orig, array.dtype, context))
         warnings.warn(msg, DataConversionWarning)
+
+    # Indices Datatype regulation
+    _check_large_sparse(array, accept_large_sparse)
```

Comment:
Is there a reason we are not just doing this in `_ensure_sparse_format` called above?

Reply:
there is no reason,

I shifted there but below some of the functions, because those function may change the indices type of spmatrix

Reply:
Really? Where do you mean? If it's possible to move it to `_ensure_sparse_format` I'd much rather it